### ISSUE TITLE
Set authorityKeyIdentifier=keyid if Authority Key Identifier is set in the UI

### ISSUE THREAD

**Issue by sebschrader** [chris2511/xca#1]
Most certificates out there don't include the `issuer` in the Authority Key Identifier, they only include the `keyid`. In fact Mozilla's CA guidelines explicitly forbid such a practice. See [section 5.2](https://www.mozilla.org/en-US/about/governance/policies/security-group/certs/policy/).

Naturally I couldn't find any well-known public site (e.g. google.com, microsoft, letsencrypt.org), where any of the certificates in the chain included something other than the `keyid` in the AKI.

XCA however sets `authorityKeyIdentifier=keyid:always,issuer:always`, if one ticks _Authority Key Identifier_ in the _Extensions_ tab. A different behavior can only be accomplished using the _Advanced_ tab.

What's even worse: If one uses the _Transform_ → _Similar Certificate_ feature on a certificate, that had only `keyid` as `authorityKeyIdentifier`, XCA drops this information and sets its AKI option in the `Extensions` tab. One has to manually go back to the `Advanced` tab and enter `authorityKeyIdentifier=keyid` again and unset the checkbox in the `Extensions` tab. This can be easily be forgotten.

Please change the default `authorityKeyIdentifier` setting to `keyid`, if the AKI is selected in the UI or at least make it configurable.

**Comment by chris2511** [comment]
Hi Sebastian,

this has already been changed for XCA 1.4.1 with commit e3c9d7bff84f54f9a2cccd96804a9964419439b7
The pre-version 1.4.1-pre02 containing this change is available at https://hohnstaedt.de/downloads/
If you also think this is what you asked for (and probably checked with 1.4.1-pre02) please close this issue.

**Comment by sebschrader** [comment]
Hi Christian,

thank you very much for your fast response. I just checked a local build of 1.4.1-pre02 and it worked as expected.

Just for the record: My previous statement was not entirely correct. As can be seen from the commit you referenced, XCA used to set `authorityKeyIdentifier=keyid:always,issuer:always`. I updated my initial comment accordingly.

Recent versions of OpenSSL just ignore `issuer`, if not suffixed with `always`, to comply with PKIX recommendations and CA guidelines. Is there any reason, why you left `issuer` in there and not set `authorityKeyIdentifier=keyid`?
